core/src/main/java/io/undertow/protocols/http2/HpackDecoder.java

```diff
@@ -260,12 +258,12 @@ private String readHpackString(ByteBuffer buffer) throws HpackException {
     }
 
     private String readHuffmanString(int length, ByteBuffer buffer) throws HpackException {
+        StringBuilder stringBuilder = new StringBuilder(length);
```

Comment:
final ?

Reply:
In this fix, I referenced the patch for a similar vulnerability (CVE-2020-17527) in the corresponding functional module of Tomcat, as shown in this commit: https://github.com/apache/tomcat/commit/d56293f816d6dc9e2b47107f208fa9e95db58c65#diff-98723542790b05d2e3c2dadeef7c632e641e3118b3b809660e9b1ef41686cfe0R226. I removed the `final` from the `StringBuilder stringBuilder` declaration, enhancing code flexibility and allowing for future reassignment if needed.

Certainly, I think it is also acceptable to retain the `final` modifier.

Reply:
Oh, I’m sorry, I accidentally deleted a previous conversation.